common/evmcompat/helpers.go

```diff
@@ -156,15 +156,19 @@ func recoverAddressFromEos(hash []byte, sig []byte) (common.Address, error) {
 
 func recoverAddressFromEosScatter(hash []byte, sig []byte) (common.Address, error) {
 	var signer common.Address
-	if len(sig) != 67 {
-		return signer, fmt.Errorf("eos signature must be 67 bytes, not %d bytes", len(sig))
+
+	if len(sig) != 107 {
+		return signer, fmt.Errorf("eos scatter signature must be 107 bytes, not %d bytes", len(sig))
 	}
-	signature := ecc.NewSigNil()
-	_, err := signature.Unpack(sig[1:])
+	signature, err := ecc.NewSignature(string(sig[6:]))
 	if err != nil {
 		return signer, fmt.Errorf("cannot unpack eos signature %v", string(sig))
 	}
-	pubKey, err := signature.PublicKey(hash)
+
+	nonceHash := sha256.Sum256([]byte(hex.EncodeToString(sig[:6])))
```

Comment:
sig[1:7] is already the hash of the nonce. So don't we want `nonceHash  = sig[1:7]`?